### ISSUE TITLE
`dbtcloud_databricks_credential` should work with `dbtcloud_global_connection` that doesn't generate an `adapter_id`

### ISSUE THREAD

**Issue by Stevedow99** [dbt-labs/terraform-provider-dbtcloud#306]
## Describe the bug
The resource`dbtcloud_databricks_credential` requires `adapter_id` but the `dbtcloud_global_connection` doesn't return `adapter_id`

## Resource configuration
```
resource "dbtcloud_databricks_credential" "my_databricks_cred" {
  project_id   = dbtcloud_project.dbt_project.id

  adapter_id   = dbtcloud_connection.my_databricks_connection.adapter_id <------- need this to get returned from dbtcloud_global_connection if the connection type is Databricks


  target_name  = "prod"
  token        = "abcdefgh"
  schema       = "my_schema"
  adapter_type = "databricks"
}


```

## Expected behavior
I would expect `adapter_id` be returned when creating the resource `dbtcloud_global_connection` if the global connection is of type Databricks

## Config (please complete the following information):
( the version can be retrieved running the command `terraform providers` )
 - dbt Cloud provider version: 0.3.18

## Additional context
Suitable work around for the time being using the http provider 
```
provider "http" {}

data "http" "dbt_cloud_databricks_global_conn" {
  url = "https://cloud.getdbt.com/api/v3/accounts/${var.dbt_account_id}/connections/${var.dbt_dbx_global_connection_id}"
  
  request_headers = {
    Authorization = "Token ${var.dbt_token}"
  }
}



output "dbt_cloud_databricks_global_conn_adapter_id" {
  value = jsondecode(data.http.dbt_cloud_databricks_global_conn.response_body).data.config.adapter_id
}

```

**Comment by b-per** [comment]
I need to do a few more checks but I just think that connections created as global connections don't have any `adapter_id` anymore (the ones created as non-global connections and then migrated to global still do).

So, the solution wouldn't be to add `adapter_id` to the global connection, but to remove it as a required field for the `..._credentials` resources.

In your example where you got an `adapter_id`, was this connection created from scratch as a `dbtcloud_global_connection` or was it created as a `dbtcloud_connection`?

**Comment by b-per** [comment]
After some analysis, the problem might require some significant work as we completely changed our API for creating credentials when using global connections.

This is not a quick fix unfortunately so it will take longer than I expected.

**Comment by b-per** [comment]
Should be fixed in 0.3.19
You just should not provide an `adapter_id` anymore when using global connections.
